service/Abstractions/ContentStorage/IContentStorage.cs

```diff
@@ -90,4 +90,20 @@ public interface IContentStorage
         string fileName,
         bool logErrIfNotFound = true,
         CancellationToken cancellationToken = default);
+
+    /// <summary>
+    /// Fetch file info from storage
+    /// </summary>
+    /// <param name="index">Index name</param>
+    /// <param name="documentId">Document ID</param>
+    /// <param name="fileName"></param>
+    /// <param name="logErrIfNotFound">Whether to log an error if the file does not exist. An exception will be raised anyway.</param>
+    /// <param name="cancellationToken">Async task cancellation token</param>
+    /// <returns>File content</returns>
+    Task<StreamableContentFile> GetFileInfoAsync(
```

Comment:
the method is not fetching just the info but also the content, so I think we should consider reusing the existing `ReadFileAsync` method, e.g. allowing streaming and including the data needed. Or, we can limit `GetFileInfoAsync` to return just info and not the content, changing `ReadFileAsync` to return a streamable file content (without anything else).